### ISSUE TITLE
OTP Autocomplete issue (Safari)

### ISSUE THREAD

**Issue by bberz-groma** [andreaswilli/react-verification-input#115]
First off, thank you for all the work on this component - made our implementation of this feature for MFA so quick and easy.

We had a reported bug related to using the autocomplete feature on Safari (both desktop and mobile) which results in sporadic behavior / failure of code entry into the inputs. After a lot of fun, painful I mean, debugging we determined that there appeared to be a race condition with focusing on the inputs and which input was actually selected.

![Pasted_Image_4_12_23__1_17_PM](https://user-images.githubusercontent.com/100605721/231534952-8f94578f-db55-485f-b567-e6a786b0f73d.png)
![Pasted_Image_4_12_23__1_20_PM](https://user-images.githubusercontent.com/100605721/231534970-e24709fc-023d-4339-8411-961408276608.png)

it's worth noting that the code which was entered was `491879`. As you can see, the code is not in order and thus values were entered and then replaced since the input focus wasn't firing constantly. 

This was seen both on mobile and desktop safari browsers and we're trying to confirm if this was consistent on Chrome as well.

**Comment by bberz-groma** [comment]
I wanted to posted the above for record sake but we have determined what the issue was - at least for us.

The methods of `selectInputContent` and `focusInput` leverage an `requestAnimationFrame` and that method wraps the ref `input.focus` && `input.select` methods (as well as the blur method). When we removed this `requestAnimationFrame` the autocomplete works as expected and consistently. 

https://user-images.githubusercontent.com/100605721/231537421-e66c435c-cf74-4151-b494-d4c0c9e584b2.mov


![image](https://user-images.githubusercontent.com/100605721/231536564-d204bf24-3f1c-4924-90c9-73dadcf2686b.png)

![image](https://user-images.githubusercontent.com/100605721/231536598-27203857-9459-4c18-8b4d-9fb790891b31.png)


I am unsure of the benefits / tradeoffs of this utility, but it might be worth exploring removing if other people report this same issue.
